newIDE/app/src/LayersList/index.js

```diff
@@ -130,8 +130,8 @@ const LayersListBody = (props: LayersListBodyProps) => {
             }));
           } else {
             onRenameLayer(layerName, newName, doRename => {
-              if (doRename)
-                layersContainer.getLayer(layerName).setName(newName);
+              if (!doRename) return;
+              layersContainer.getLayer(layerName).setName(newName);
               gd.WholeProjectRefactorer.renameLayer(
                 project,
```

Comment:
It's worth eliminating this callback "done" I guess and have just:
```
layersContainer.getLayer(layerName).setName(newName);
gd.WholeProjectRefactorer
onLayerRenamed();
```

(take the opportunity to name this "onLayerRenamed"

Reply:
totally right, I've just done that